client/pkg/omni/resources/system/resource_labels.go

```diff
@@ -0,0 +1,72 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package system
+
+import (
+	"github.com/cosi-project/runtime/pkg/controller/generic"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/meta/spec"
+
+	"github.com/siderolabs/omni/client/pkg/omni/resources"
+)
+
+// NewResourceLabels creates new ResourceLabels state.
+func NewResourceLabels[T generic.ResourceWithRD](id string) *ResourceLabels[T] {
```

Comment:
The cool part of making this resource generic is that it's possible to use `safe.ReaderGetByID[*system.ResourceLabels[*omni.MachineStatus]](...)`.